robot/Cumulus/resources/GiftEntryPageObject.py

```diff
@@ -113,9 +98,178 @@ def store_template_record_id(self,template):
+        if not check.is_selected():
+            try:
+                self.salesforce._jsclick(field_checkbox)
+            except ElementClickInterceptedException:
+                self.selenium.execute_javascript("window.scrollBy(0,0)")
+                self.salesforce._jsclick(field_checkbox)
+        label=": "+field    
+        self.selenium.wait_until_page_contains(label)
+
+
+    @capture_screenshot_on_error                
+    def fill_template_form(self,**kwargs):
+        """Add default values to template builder form fields or set fields as required. 
+        Key is field main label name and value is again another dictionary of field type and value
+        ex: Payment: Payment Method is Main Label with Default Value as field type and Chech is value"""
+        self.selenium.execute_javascript("window.scrollBy(0,0)")
+        for field,option in kwargs.items():
+            for section,value in option.items():
+                if section=="Required":
+                    label=f'{section} {field}'
+                    if value=='checked':
+                        field_checkbox=npsp_lex_locators["gift_entry"]["field_input"].format(label,"input")
+                        self.selenium.scroll_element_into_view(field_checkbox)
+                        cb_loc=self.selenium.get_webelement(field_checkbox)
+                        if not cb_loc.is_selected():
+                            self.salesforce._jsclick(field_checkbox)
+                    elif value=='unchecked': 
+                        field_checkbox=npsp_lex_locators["gift_entry"]["field_input"].format(label,"input")
+                        self.selenium.scroll_element_into_view(field_checkbox)
+                        cb_loc=self.selenium.get_webelement(field_checkbox)
+                        if cb_loc.is_selected():
+                            self.salesforce._jsclick(field_checkbox)
+                elif section=="Default Value":
+                    key=f'{section} {field}'
+                    field_loc=npsp_lex_locators["gift_entry"]["field_input"].format(key,"input")
+                    placeholder=self.selenium.get_webelement(field_loc).get_attribute("placeholder")
+                    self.selenium.scroll_element_into_view(field_loc)
+                    try:
+                        self.selenium.click_element(field_loc)
+                    except ElementClickInterceptedException:
+                        self.selenium.execute_javascript("window.scrollBy(0,100)")
+                        self.selenium.click_element(field_loc)    
+                    if placeholder=="Select an Option":
+                        popup=npsp_lex_locators["flexipage-popup"]
+                        self.selenium.wait_until_page_contains_element(popup)
+                        option=npsp_lex_locators["span_button"].format(value)
+                        self.selenium.click_element(option)
+                    elif placeholder=="Search...":
```

Comment:
Does the string always has "Search..." ? It would be a nice way of comparing  if we can use contains say in case if there are no trailing dots or only two dots after "Search"